src/api/v2/quickPickWizard/GenericQuickPickStep.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, NoResourceFoundError } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ResourceModelBase } from '../v2AzureResourcesApi';
+import { ContextValueFilter, matchesContextValueFilter } from './ContextValueFilter';
+import { QuickPickWizardContext } from './QuickPickWizardContext';
+
+export class GenericQuickPickStep<TModel extends ResourceModelBase> extends AzureWizardPromptStep<QuickPickWizardContext<TModel>> {
+    public supportsDuplicateSteps = true;
+
+    public constructor(
+        protected readonly treeDataProvider: vscode.TreeDataProvider<TModel>,
+        protected readonly contextValueFilter: ContextValueFilter,
+    ) {
+        super();
+    }
+
+    public async prompt(wizardContext: QuickPickWizardContext<TModel>): Promise<void> {
+        const selected = await wizardContext.ui.showQuickPick(await this.getPicks(wizardContext), { /* TODO: options */ });
+        wizardContext.currentNode = selected.data;
+    }
+
+    public shouldPrompt(_wizardContext: QuickPickWizardContext<TModel>): boolean {
+        return true;
+    }
+
+    protected async getPicks(wizardContext: QuickPickWizardContext<TModel>): Promise<IAzureQuickPickItem<TModel>[]> {
+        const children = (await this.treeDataProvider.getChildren(wizardContext.currentNode)) || [];
+
+        const matchingChildren = children.filter(c => matchesContextValueFilter(c, this.contextValueFilter));
+        const nonLeafChildren = children.filter(c => c.quickPickOptions?.isLeaf === false);

```

Comment:
nit: Don't need to run this until after check of `matchingChildren.length === 0`